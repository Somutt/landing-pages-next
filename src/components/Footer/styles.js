import styled, { css } from 'styled-components';
import { Wrapper as TextComponent } from '../TextComponent/styles';
import { Wrapper as SectionContainer } from '../SectionContainer/styles';

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    ${TextComponent} {
      font-size: ${theme.font.sizes.xsmall};
    }

    ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
