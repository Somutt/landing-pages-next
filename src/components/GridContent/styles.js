import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `};
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.big} 0;
  `};
`;
