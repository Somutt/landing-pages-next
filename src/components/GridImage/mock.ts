import { GridImageProps } from '.';

export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      altText: 'Teste 1',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
} as GridImageProps;
