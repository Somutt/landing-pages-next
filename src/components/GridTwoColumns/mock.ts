import { GridTwoColumnsProps } from '.';

export default {
  title: 'Grid two columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit.
    Nisi dicta commodi numquam ad provident ut expedita placeat illum facilis omnis
    necessitatibus doloribus sunt beatae dolorem, maiores eos veritatis!
    Beatae, obcaecati.`,
  imgSrc: 'assets/images/backstrapi.svg',
} as GridTwoColumnsProps;
