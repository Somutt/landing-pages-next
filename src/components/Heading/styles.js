import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
`;

export const Title = styled.h1`
  ${({ theme, dark, size, uppercase }) => css`
    color: ${dark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
