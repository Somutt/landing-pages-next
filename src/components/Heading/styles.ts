import styled, { css } from 'styled-components';
import { DefaultTheme } from 'styled-components/dist/types';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
};

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, dark, size, uppercase }) => css`
    color: ${dark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
