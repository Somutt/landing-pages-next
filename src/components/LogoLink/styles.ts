import styled, { css } from 'styled-components';

export const Wrapper = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
      height: 2rem;
    }
  `};
`;
