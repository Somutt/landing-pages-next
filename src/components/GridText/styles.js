import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.big};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.big};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.large};
      top: -1.5rem;
      left: -4rem;
      transform: rotate(10deg);
    }
  `}
`;
