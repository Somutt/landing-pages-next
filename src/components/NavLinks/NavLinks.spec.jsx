import { screen } from '@testing-library/react';
import { NavLinks } from '.';

import { renderTheme } from '../../utils/renderTheme';
import mock from './mock';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
  it('should render all MenuLinks correctly', () => {
    renderTheme(<NavLinks links={mock}></NavLinks>);
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLinks></NavLinks>);
    const links = screen.queryAllByText(/link/i);
    expect(links).toHaveLength(0);
  });

  it('should render all MenuLinks correctly', () => {
    renderTheme(<NavLinks links={mock}></NavLinks>);
    const link = screen.getByText(/link 10/i);
    expect(link.parentElement).toHaveStyleRule('flex-flow', 'column wrap', {
      media: theme.media.lteMedium,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock}></NavLinks>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
