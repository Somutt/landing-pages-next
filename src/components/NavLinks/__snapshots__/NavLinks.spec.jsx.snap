// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c0 {
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 0.8rem;
  color: #0A1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {
  .c0 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="target 1"
    target="_self"
  >
    link 1
  </a>
  <a
    class="c1"
    href="target 2"
    target="_self"
  >
    link 2
  </a>
  <a
    class="c1"
    href="target 3"
    target="_self"
  >
    link 3
  </a>
  <a
    class="c1"
    href="target 4"
    target="_self"
  >
    link 4
  </a>
  <a
    class="c1"
    href="target 5"
    target="_self"
  >
    link 5
  </a>
  <a
    class="c1"
    href="target 6"
    target="_self"
  >
    link 6
  </a>
  <a
    class="c1"
    href="target 7"
    target="_self"
  >
    link 7
  </a>
  <a
    class="c1"
    href="target 8"
    target="_self"
  >
    link 8
  </a>
  <a
    class="c1"
    href="target 9"
    target="_self"
  >
    link 9
  </a>
  <a
    class="c1"
    href="target 10"
    target="_self"
  >
    link 10
  </a>
</nav>
`;
